tools/annotato/src/boundingbox.rs

```diff
@@ -0,0 +1,165 @@
+        }
+    }
+
+    pub fn set_opposing_corner(&mut self, plot_bottom_right: PlotPoint) {
+        self.opposing_corner = plot_bottom_right;
+    }
+
+    pub fn rect(&self) -> Rect {
+        let to_pos2 = |point: PlotPoint| Pos2::new(point.x as f32, point.y as f32);
+        Rect::from_points(&[to_pos2(self.corner), to_pos2(self.opposing_corner)])
+    }
+
+    pub fn top_left(&self) -> PlotPoint {
+        let x1 = self.corner.x;
+        let y1 = self.corner.y;
+        let x2 = self.opposing_corner.x;
+        let y2 = self.opposing_corner.y;
+
+        PlotPoint::new(x1.min(x2), y1.max(y2))
+    }
+
+    pub fn contains(&self, mouse_position: PlotPoint) -> bool {
+        let x1 = self.corner.x;
+        let y1 = self.corner.y;
+        let x2 = self.opposing_corner.x;
+        let y2 = self.opposing_corner.y;
+
+        (x1.min(x2)..=x1.max(x2)).contains(&mouse_position.x)
+            && (y1.min(y2)..=y1.max(y2)).contains(&mouse_position.y)
+    }
+
+    pub fn clip_to_image(&mut self) {
+        let x1 = self.corner.x;
+        let y1 = self.corner.y;
+        let x2 = self.opposing_corner.x;
+        let y2 = self.opposing_corner.y;
+
+        self.corner = PlotPoint::new(x1.clamp(0., 640.), y1.clamp(0., 480.));
+        self.opposing_corner = PlotPoint::new(x2.clamp(0., 640.), y2.clamp(0., 480.));
+    }
+
+    pub fn is_valid(&self) -> bool {
+        let rect = self.rect();
+        rect.area() >= 4.0
+    }
+
+    pub fn iou(&self, other: &BoundingBox) -> f32 {
+        let this_rect = self.rect();
+        let other_rect = other.rect();
+
+        let intersection = this_rect.intersect(other_rect).area();
+        let union = this_rect.area() + other_rect.area() - intersection;
+
+        intersection / union
+    }
+
+    fn closest_and_furthest_corner(&self, position: PlotPoint) -> (PlotPoint, PlotPoint) {
+        let rect = self.rect();
+        let position = Pos2::new(position.x as f32, position.y as f32);
+
+        let corners = [
+            rect.left_bottom(),
+            rect.left_top(),
+            rect.right_top(),
+            rect.right_bottom(),
+        ];
+        let closest = corners
+            .iter()
+            .min_by(|corner1, corner2| {
+                corner1
+                    .distance_sq(position)
+                    .total_cmp(&corner2.distance_sq(position))
+            })
+            .unwrap();
+        let furthest = corners
+            .iter()
+            .max_by(|corner1, corner2| {
+                corner1
+                    .distance_sq(position)
+                    .total_cmp(&corner2.distance_sq(position))
+            })
+            .unwrap();
+
+        (
+            PlotPoint::new(closest.x as f64, closest.y as f64),
+            PlotPoint::new(furthest.x as f64, furthest.y as f64),
+        )
+    }
+
+    pub fn get_closest_corner(&self, position: PlotPoint) -> PlotPoint {
+        let (closest, _) = self.closest_and_furthest_corner(position);
+        closest
```

Comment:
```suggestion
        self.closest_and_furthest_corner(position).0
```
?

or remove the tuple entirely

Reply:
uses `.0` now